devito/types/petsc.py

```diff
@@ -247,3 +247,24 @@ def solver_parameters(self):
         return self._solver_parameters
 
     func = Reconstructable._rebuild
+
+
+class PETScStruct(CompositeObject):
```

Comment:
placing this question here, but I should have asked time ago. Do we actually need `types/petsc`?

I'm now thinking PETSc support might deserve its own folder in the root directory, alongside `ir`, `types`, `passes`, `mpi`, etc

I'd prefer it centralised rather than distributed, but perhaps it's just me, thoughts @ZoeLeibowitz @mloubout @EdCaunt